### ISSUE TITLE
Minor modification in Production rule 190

### ISSUE THREAD

**Issue by vijayphoenix** [yaml/yaml-spec#34]
1. YAML specs does not explicitly mention `/* Not followed by an ns-char */` constraint in the production rule [190](https://yaml.org/spec/1.2/spec.html#c-l-block-map-explicit-key(n)), that is 

Rule mentioned
```
[190]  c-l-block-map-explicit-key(n)  ::=  “?” s-l+block-indented(n,block-out)
```
Modified Rule
```
[190]  c-l-block-map-explicit-key(n)  ::=  “?”   /* Not followed by an ns-char */  s-l+block-indented(n,block-out)
```
2. This constraint is necessary to satisfy both [Rule 126](https://yaml.org/spec/1.2/spec.html#ns-plain-first(c)) and [Rule 190](https://yaml.org/spec/1.2/spec.html#c-l-block-map-explicit-key(n)).
3. This constraint is mentioned in an almost similar [rule 184](https://yaml.org/spec/1.2/spec.html#c-l-block-seq-entry(n))
4. Modifing this rule leads to performance improvement on [Yaml Test Suite](https://github.com/yaml/yaml-test-suite), see [HsYAML PR#10](https://github.com/haskell-hvr/HsYAML/pull/10#issuecomment-493558403)

**Comment by ingydotnet** [comment]
@vijayphoenix good call. Also quote useful in several other productions. See #189.

**Comment by ingydotnet** [comment]
This was addressed in the recent 1.2.2 spec https://yaml.org/spec/1.2.2/#rule-c-l-block-map-explicit-key

184 https://yaml.org/spec/1.2.2/#rule-c-l-block-seq-entry has an explicit lookahead but we decided to make many others be just comments.
